mtc.go

```diff
@@ -1527,12 +1482,43 @@ func NewMerkleTreeProof(batch *Batch, index uint64, path []byte) *MerkleTreeProo
 	}
 }
 
-// A Trust Anchor Identifier (TAI) is used to identify a CA, or a specific batch.
+// A TrustAnchorIdentifier (TAI) is used to identify a CA, or a specific batch.
 //
 // TAI are OIDs relative to the Private Enterprise Numbers (PEN)
-// arc  1.3.6.1.4.1.
+// arc 1.3.6.1.4.1.
 type TrustAnchorIdentifier []byte
 
+func (tai *TrustAnchorIdentifier) ProofType() ProofType {
+	// TODO make this dependent on the OID
```

Comment:
We can't see it from the OID itself: we need some lookup. A caller will have a list of known CAs that include OID, public key, and proof type. The question is how to do this elegantly in the API. Perhaps we have a MTCContext struct, which when created can be passed with a `CAStore` interface that has a OID -> CAParams function.

Reply:
I've built a simple construction, such that the CA params essentially are the only entry in the 'root store'. Let me know if you would like something more elaborate, or feel free to change it yourself.